mythril/laser/plugin/plugins/plugin_annotations.py

```diff
@@ -42,18 +48,31 @@ def __copy__(self):
 
     def get_storage_write_cache(self, iteration: int):
         if iteration not in self.storage_written:
-            self.storage_written[iteration] = []
+            self.storage_written[iteration] = set()
 
         return self.storage_written[iteration]
 
     def extend_storage_write_cache(self, iteration: int, value: object):
         if iteration not in self.storage_written:
-            self.storage_written[iteration] = [value]
+            self.storage_written[iteration] = {value}
         elif value not in self.storage_written[iteration]:
-            self.storage_written[iteration].append(value)
+            self.storage_written[iteration].add(value)
+
+    def check_merge_annotation(self, other):
+        return self.has_call == other.has_call and self.path == other.path
+
+    def merge_annotation(self, other: "DependencyAnnotation"):
```

Comment:
What do you think about having merge_annotation create a new annotation? Vs mutating the current one?